### ISSUE TITLE
🚀 Feature: Connect to an external data source

### ISSUE THREAD

**Issue by SOVLOOKUP** [appwrite/appwrite#2026]
### 🔖 Feature description

I need to connect to my existing database such as  `couchdb` `pgsql`. Like [budibase](https://budibase.com/).

![pic](https://www.budibase.com/img/img/external-data-sources.webp)

### 🎤 Pitch

Its improve the portability.

Most businesses and developers consider costs of migration before using AppWrite. Data is the most important resource.

It also makes it easier for developers to migrate to AppWrite from other architectures.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by nev-21** [comment]
Yeah this is specially helpful when we need anti-corruption layers (for legacy systems) or integrations to parallel systems. 

The functions and sdk should allow to connect an external db (non appwrite db) or a rest-api to make integrations possible.

**Comment by eldadfux** [comment]
This can be achieved by self-hosting Appwrite and connecting directly to the under the hood database.
